pkg/bytecode/compiler.go

```diff
@@ -0,0 +1,120 @@
+package bytecode
+
+import (
+	"errors"
+	"fmt"
+
+	"evylang.dev/evy/pkg/parser"
+)
+
+// ErrUndefinedVar is returned when a variable name cannot
+// be resolved in the symbol table.
+var ErrUndefinedVar = errors.New("undefined variable")
+
+// Compiler is responsible for turning a parsed evy program into
+// bytecode.
+type Compiler struct {
+	constants    []value
+	instructions Instructions
+	symbolTable  *SymbolTable
+}
+
+// Bytecode represents raw evy bytecode.
+type Bytecode struct {
```

Comment:
`Bytecode` is a little stuttery given the name of the package (`bytecode`), but I admit after giving it a lot of thought, I couldn't come up with another name.